test/validation/api/pool/pool.c

```diff
@@ -1210,19 +1222,34 @@ static void pool_test_pool_statistics(odp_pool_type_t pool_type)
 			else
 				num_fails++;
 
+			stats.thread.first = first;
+			stats.thread.last = last;
 			CU_ASSERT_FATAL(odp_pool_stats(pool[i], &stats) == 0);
 			CU_ASSERT(stats.available <= num_obj - num_events);
 			CU_ASSERT(stats.cache_available <= num_obj - num_events);
+
+			for (int k = 0; k < ODP_POOL_MAX_THREAD_STATS; k++) {
+				CU_ASSERT(stats.thread.cache_available[k] <= num_obj - num_events);
+				total_cached += stats.thread.cache_available[k];
+			}
+			if (supported.bit.cache_available && supported.bit.thread_cache_available)
+				CU_ASSERT(stats.cache_available == total_cached);
```

Comment:
This will not hold if odp_thread_count_max() > ODP_POOL_MAX_THREAD_STATS

Reply:
Yes, but as this test is single threaded and thread ids are allocated starting from 0 this should be OK.

Reply:
The API does not say in which order thread IDs are assigned, so one cannot portably assume that the thread_id of the single thread of a single threaded app is zero or any other particular value.

I also think it is incorrect to assume that the number of running threads is relevant here. Thread local caches may get filled at ODP startup time so that the cache of a not-yet-running thread is not empty. So I think the code here should read the local cache size of all possible threads (using multiple stats calls if necessary) if it wants to check that the sum equals the total cache size.

Reply:
The thread ID allocation should be clarified in the API  as discussed in #1619. Anyway, it makes sense to test `odp_pool_stats()` with a varying thread range, I'll update the test in V5.